lib/core.ps1

```diff
@@ -604,22 +610,64 @@ set args=%args:(=``(%
+where /q pwsh.exe
+if %errorlevel% equ 0 (
+    pwsh -noprofile -ex unrestricted `"& '$resolved_path' $arg %args%;exit `$lastexitcode`"
+) else (
+    powershell -noprofile -ex unrestricted `"& '$resolved_path' $arg %args%;exit `$lastexitcode`"
+)" | out-file "$shim.cmd" -encoding ascii
+
+        warn_on_overwrite $shim $path
+        "#!/bin/sh
+# $resolved_path
+if command -v pwsh.exe &> /dev/null; then
+    pwsh.exe -noprofile -ex unrestricted `"$resolved_path`" $arg `"$@`"
+else
+    powershell.exe -noprofile -ex unrestricted `"$resolved_path`" $arg `"$@`"
+fi" | out-file $shim -encoding ascii
     } elseif($path -match '\.jar$') {
-        "@java -jar `"$resolved_path`" $arg %*" | out-file "$shim.cmd" -encoding ascii
-        "#!/bin/sh`njava -jar `"$resolved_path`" $arg `"$@`"" | out-file $shim -encoding ascii
+        warn_on_overwrite "$shim.cmd" $path
+        "@rem $resolved_path
+@java -jar `"$resolved_path`" $arg %*" | out-file "$shim.cmd" -encoding ascii
+
+        warn_on_overwrite $shim $path
+        "#!/bin/sh
+# $resolved_path
+java -jar `"$resolved_path`" $arg `"$@`"" | out-file $shim -encoding ascii
+    } elseif($path -match '\.py$') {
+        warn_on_overwrite "$shim.cmd" $path
+        "@rem $resolved_path
+@python `"$resolved_path`" $arg %*" | out-file "$shim.cmd" -encoding ascii
+
+        warn_on_overwrite $shim $path
+        "#!/bin/sh
+# $resolved_path
+python `"$resolved_path`" $arg `"$@`"" | out-file $shim -encoding ascii
+    } else {
+        warn_on_overwrite "$shim.cmd" $path
+        # find path to Git's bash so that batch scripts can run bash scripts
+        $gitdir = (Get-Item (Get-Command git -ErrorAction:Stop).Source -ErrorAction:Stop).Directory.Parent
+        if ($gitdir.FullName -imatch 'mingw') {
+          $gitdir = $gitdir.Parent
+        }
+        "@rem $resolved_path
+@`"$(Join-Path (Join-Path $gitdir.FullName 'bin') 'bash.exe')`" `"$resolved_path`" $arg %*" | out-file "$shim.cmd" -encoding ascii
+
+        warn_on_overwrite $shim $path
+        "#!/bin/sh
+# $resolved_path
+`"$resolved_path`" $arg `"$@`"" | out-file $shim -encoding ascii
     }
 }
 
 function get_shim_path() {
-    $shim_path = "$(versiondir 'scoop' 'current')\supporting\shimexe\bin\shim.exe"
+    $shim_path = "$(versiondir 'scoop' 'current')\supporting\shims\kiennq\shim.exe"
     $shim_version = get_config 'shim' 'default'
     switch ($shim_version) {
         '71' { $shim_path = "$(versiondir 'scoop' 'current')\supporting\shims\71\shim.exe"; Break }
-        'kiennq' { $shim_path = "$(versiondir 'scoop' 'current')\supporting\shims\kiennq\shim.exe"; Break }
+        'scoopcs' { $shim_path = "$(versiondir 'scoop' 'current')\supporting\shimexe\bin\shim.exe"; Break }
         'rshim' { $shim_path = "$(versiondir 'scoop' 'current')\supporting\shims\rshim\shim.exe"; Break }
+        'kiennq' { Break } # for backward compatibility
```

Comment:
If the user has kiennq set, he will get scoop's default shim, not kiennq, now?

Reply:
No, if the user has kiennq set, it will use the path `"$(versiondir 'scoop' 'current')\supporting\shims\kiennq\shim.exe"` (and come out of the switch case)